azurerm/internal/services/databasemigration/tests/resource_arm_database_migration_project_test.go

```diff
@@ -0,0 +1,206 @@
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_database_migration_project", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDatabaseMigrationProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDatabaseMigrationProject_basic(data),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMDatabaseMigrationProject_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMDatabaseMigrationProject_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_database_migration_project", "test")
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDatabaseMigrationProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDatabaseMigrationProject_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDatabaseMigrationProjectExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMDatabaseMigrationProject_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDatabaseMigrationProjectExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.name", "test"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMDatabaseMigrationProjectExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Database Migration Project not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serviceName := rs.Primary.Attributes["service_name"]
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).DatabaseMigration.ProjectsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := client.Get(ctx, resourceGroup, serviceName, name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Database Migration Project (Project Name %q / Service Name %q / Group Name %q) does not exist", name, serviceName, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on ProjectsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMDatabaseMigrationProjectDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).DatabaseMigration.ProjectsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_database_migration_project" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serviceName := rs.Primary.Attributes["service_name"]
```

Comment:
could we use the parse ID function here like in #5779

Reply:
I think i may have missed this in the service PR

Reply:
The service PR has handled that, but this PR missed that. This resource actually has been implemented for months, and I just pick it up after you merged the service one.
I should have come up with it. Sorry for that!